### ISSUE TITLE
JulianDate.fromIso8601 throws error

### ISSUE THREAD

**Issue by speigg** [AnalyticalGraphicsInc/cesium#2544]
```javascript
var date = new Cesium.JulianDate()
var isoDate = Cesium.JulianDate.toIso8601() //  "-4713-11-24T12:00:00Z"
Cesium.JulianDate.fromIso8601(date) // throws DeveloperError "Invalid ISO 8601 date."
```

**Comment by mramato** [comment]
Thanks, it is an invalid ISO8601 date, but we should throw in `toIso8601` indicating why instead of in `fromIso8601`.

**Comment by speigg** [comment]
What makes it and invalid ISO8601 date?

**Comment by mramato** [comment]
According to the spec, 0000 and 9999 are the min/max values for the year components, so `0000-01-01T00:00:00Z` is the earliest date possible.

That being said, the spec makes allowances for "expanded representation", which would allow Cesium to support an arbitrary number of digits for the year if the need every arises.

**Comment by speigg** [comment]
Okay thanks for explaining. Shouldn't the default JulianDate be `000-01-01T00:00:00Z` if that's the case?

**Comment by mramato** [comment]
No, `JulianDate` is different format, [Julian day](http://en.wikipedia.org/wiki/Julian_day).  A Julian day of 0 corresponds to -4713 BC.  There are lots of ways to represent dates and they are not all compatible.  `JulianDate` happens to be the most precise, accurate and performant for Cesium's use cases, so that's what we use internally.  We provide utility functions to go back and forth in different formats, but that does not mean all formats are compatible.  For example, JavaScript `Date` objects can't represent fractional milliseconds, so you lose precision when going from `JulianDate` to `Date`
